tests/data/dataset_test.py

```diff
@@ -38,9 +38,9 @@ def test_as_arrays(self):
         dataset.index_instances(self.vocab)
         padding_lengths = dataset.get_padding_lengths()
         arrays = dataset.as_arrays(padding_lengths)
+        text1 = arrays["text1"]["words"]
```

Comment:
I kind of disagree - I think it's helpful to have tests which show how some features are connected together and this is well within the realm of what we should be able to do with a `Vocab`, `Dataset` and `TokenIndexer`. Also I don't exactly understand what you mean about how this gives the wrong impression of how to access the arrays?